### PR TITLE
tests: Change duplicated names of the test cases

### PR DESCRIPTION
**Pull request by katgiadla**
Some names of the test cases are duplicated within the project.
This commit contains the proposed names of the test scenarios.

Signed-off-by: Katarzyna Giadla <katarzyna.giadla@nordicsemi.no>

### CODE REVIEW

File: samples/boards/esp32/spiram_test/sample.yaml [hunk 1]
```diff
@@ -2,6 +2,6 @@ sample:
   description: Application to test memory allocation from SPIRAM
   name: spiram_test
 tests:
-  sample.board.esp32:
+  esp32.spiram_test:
```
Comment (nashif) [inline samples/boards/esp32/spiram_test/sample.yaml:5]: start all samples with sample.

File: tests/drivers/counter/counter_basic_api/testcase.yaml [hunk 2]
```diff
@@ -1,11 +1,11 @@
 tests:
-  drivers.counter:
+  drivers.basic_api.counter:
```
Comment (nashif) [inline tests/drivers/counter/counter_basic_api/testcase.yaml:2]: drivers.counter.basic_api

File: tests/drivers/counter/counter_basic_api/testcase.yaml [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
     tags: drivers
     depends_on: counter
     min_ram: 16
     platform_exclude: nucleo_f302r8
     timeout: 600
-  drivers.counter_nrf_zli:
+  drivers.basic_api.counter_nrf_zli:
```
Comment (nashif) [inline tests/drivers/counter/counter_basic_api/testcase.yaml:8]: drivers.counter.....

File: tests/drivers/counter/counter_nrf_rtc/fixed_top/testcase.yaml [hunk 4]
```diff
@@ -1,5 +1,5 @@
 tests:
-  drivers.counter:
+  drivers.counter_nrf_rtc.counter:
```
Comment (nashif) [inline tests/drivers/counter/counter_nrf_rtc/fixed_top/testcase.yaml:2]: drivers.counter.nrf_rtc:

File: tests/drivers/dac/dac_api/testcase.yaml [hunk 5]
```diff
@@ -1,7 +1,7 @@
 common:
   tags: dac drivers userspace
 tests:
-  drivers.dac:
+  drivers.dac_api:
```
Comment (nashif) [inline tests/drivers/dac/dac_api/testcase.yaml:4]: drivers.dac.api

File: tests/drivers/dac/dac_loopback/testcase.yaml [hunk 6]
```diff
@@ -1,7 +1,7 @@
 common:
   tags: dac drivers
 tests:
-  drivers.dac:
+  drivers.dac_loopback:
```
Comment (nashif) [inline tests/drivers/dac/dac_loopback/testcase.yaml:4]: drivers.dac.loopback

File: tests/drivers/i2c/i2c_api/testcase.yaml [hunk 7]
```diff
@@ -1,5 +1,5 @@
 tests:
-  drivers.i2c:
+  drivers.i2c_api:
```
Comment (nashif) [inline tests/drivers/i2c/i2c_api/testcase.yaml:2]: drivers.i2c.api

File: tests/drivers/uart/uart_pm/testcase.yaml [hunk 8]
```diff
@@ -46,7 +46,7 @@ tests:
       - CONFIG_UART_ASYNC_API=n
       - CONFIG_UART_0_ENHANCED_POLL_OUT=y
 
-  drivers.uart_async:
```
Comment (nashif) [inline tests/drivers/uart/uart_pm/testcase.yaml:49]: drivers.uart.async

**Review by nashif — CHANGES_REQUESTED** [review]

File: samples/boards/esp32/spiram_test/sample.yaml (hunk 1 above)
Comment (katgiadla) [reply samples/boards/esp32/spiram_test/sample.yaml]: done

File: tests/drivers/counter/counter_basic_api/testcase.yaml (hunk 2 above)
Comment (katgiadla) [reply tests/drivers/counter/counter_basic_api/testcase.yaml]: done

File: tests/drivers/counter/counter_nrf_rtc/fixed_top/testcase.yaml (hunk 4 above)
Comment (katgiadla) [reply tests/drivers/counter/counter_nrf_rtc/fixed_top/testcase.yaml]: done

File: tests/drivers/dac/dac_api/testcase.yaml (hunk 5 above)
Comment (katgiadla) [reply tests/drivers/dac/dac_api/testcase.yaml]: done

File: tests/drivers/dac/dac_loopback/testcase.yaml (hunk 6 above)
Comment (katgiadla) [reply tests/drivers/dac/dac_loopback/testcase.yaml]: done

File: tests/drivers/uart/uart_pm/testcase.yaml (hunk 8 above)
Comment (katgiadla) [reply tests/drivers/uart/uart_pm/testcase.yaml:49]: done

**Review by PerMac — DISMISSED** [review]

**Review by nordic-krch — DISMISSED** [review]

**Review by rlubos — DISMISSED** [review]
The net part looks fine.

**Review by nashif — CHANGES_REQUESTED** [review]

**Review by jakub-uC — DISMISSED** [review]

**Review by alwa-nordic — DISMISSED** [review]

**Review by tbursztyka — DISMISSED** [review]
once comments applied, lgtm.

File: tests/net/socket/tls_ext/testcase.yaml [hunk 9]
```diff
@@ -1,6 +1,6 @@
 common:
   tags: net socket tls
 tests:
-  net.socket.tls:
+  net.socket.tls_ext:
```
[the remark this answers is not in the document]
Comment (katgiadla) [reply tests/net/socket/tls_ext/testcase.yaml]: done

File: tests/net/socket/af_packet_ipproto_raw/testcase.yaml [hunk 10]
```diff
@@ -5,5 +5,5 @@ common:
   # needs root permission.
   platform_exclude: native_posix native_posix_64
 tests:
-  net.socket.packet:
+  net.socket.af_packet_ipproto_raw:
```
[the remark this answers is not in the document]
Comment (katgiadla) [reply tests/net/socket/af_packet_ipproto_raw/testcase.yaml]: done

File: samples/bluetooth/direction_finding_peripheral/sample.yaml [hunk 11]
```diff
@@ -2,15 +2,15 @@ sample:
   name: Direction Finding Peripheral
   description: Sample application showing peripheral role of Direction Finding in connected mode
 tests:
-  sample.bluetooth.direction_finding_connectionless_rx:
+  sample.bluetooth.direction_finding_peripheral:
```
[the remark this answers is not in the document]
Comment (katgiadla) [reply samples/bluetooth/direction_finding_peripheral/sample.yaml]: done

File: samples/bluetooth/direction_finding_central/sample.yaml [hunk 12]
```diff
@@ -2,15 +2,15 @@ sample:
   name: Direction Finding Central
   description: Sample application showing central role of Direction Finding in connected mode
 tests:
-  sample.bluetooth.direction_finding_connectionless_rx:
+  sample.bluetooth.direction_finding_central:
```
[the remark this answers is not in the document]
Comment (katgiadla) [reply samples/bluetooth/direction_finding_central/sample.yaml]: done

**Review by dcpleung — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Review by PerMac — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**